File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PersonEntityImpl.java
package com.github.chojmi.inspirations.data.entity.photos;

import android.os.Parcelable;

import com.github.chojmi.inspirations.data.entity.helpers.ImageLinkProvider;
import com.github.chojmi.inspirations.domain.entity.people.PersonEntity;
import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

@AutoValue
public abstract class PersonEntityImpl implements Parcelable, PersonEntity {
    public static TypeAdapter<PersonEntityImpl> typeAdapter(Gson gson) {
        return new AutoValue_PersonEntityImpl.GsonTypeAdapter(gson);
    }

    @Override
    public abstract String getUsername();

    abstract String getNsid();

    @SerializedName("iconserver")
    abstract int getServer();

    @SerializedName("iconfarm")
    abstract int getFarm();

    @Override
    public String getIconUrl() {
        return ImageLinkProvider.provideLink(getFarm(), getServer(), getNsid());
    }
}

File: data/src/main/java/com/github/chojmi/inspirations/data/entity/photos/PersonEntityImpl.kt
package com.github.chojmi.inspirations.data.entity.photos

import com.github.chojmi.inspirations.data.entity.helpers.ImageLinkProvider
import com.github.chojmi.inspirations.domain.entity.people.PersonEntity
import com.google.gson.annotations.SerializedName

data class PersonEntityImpl(private val username: String,
                            private val nsid: String,
                            @SerializedName("iconserver") private val server: Int,
                            @SerializedName("iconfarm") private val farm: Int) : PersonEntity {

    override fun getId(): String = nsid

    override fun getUsername(): String = username

    override fun getIconUrl(): String = ImageLinkProvider.provideLink(farm, server, nsid)
}